src/behaviors/Behavior.js

```diff
@@ -6,7 +6,7 @@ var Base = require('../lib/Base');
 
 var Column = require('./Column');
 var dialogs = require('../dialogs');
-var CellProvider = require('../lib/CellProvider');
+var CellProvider = require('../cellRenderers/CellProvider');
```

Comment:
I completely agree and that is what I meant by "reorganize." Like the cellEditors folder, we would have a base class, etc.

The current CellProvider code should be made a base class which we should rename `CellRenderer`. Then each cell renderer should extend that class. Any currently "inner" functions needed by more than one renderer should be moved to the prototype.

This would let the app developer "require" only the renderers he needs. Which is cool.

We also need to get away form the index.js thing where we "pre-register" all of them just so they can be randomly referenced by name later. Rather, we should force app developer to "require" only the ones he's using, and register them himself.

When do you think we should do this if not now? I am only saying this because looks like we're putting them off till end of week anyway so _we have the time._ AND I'd hate to see William start making cell renderers to the existing structure only to have to refactor his work in a few weeks. Although not a biggie, I suppose. It would be nice to smooth out the remaining wrinkles, like the crazy structures here. (Like, what does "cellCache" mean anyway?)